### PR TITLE
fix(node): Mempool Events Not Firing While Syncing

### PR DESCRIPTION
**Pull request by micahriggan**
* Overwriting batches for mempool transactions to do setOnInsert
* This allows us to handle mempool transactions while syncing
* Added tests for pruneMempool and events while syncing

### CODE REVIEW

File: packages/bitcore-node/src/decorators/Loggify.ts [hunk 1]
```diff
@@ -10,28 +10,26 @@ export function SavePerformance(logPrefix, startTime, endTime) {
...
-      avg: totalTime
+      avg: totalTime,
+      max: totalTime
     };
   } else {
     PerformanceTracker[logPrefix].time += totalTime;
     PerformanceTracker[logPrefix].count++;
     PerformanceTracker[logPrefix].avg = PerformanceTracker[logPrefix].time / PerformanceTracker[logPrefix].count;
+    const currentMax = PerformanceTracker[logPrefix].max;
+    PerformanceTracker[logPrefix].max = totalTime > currentMax ? totalTime : currentMax;
```
Comment (micahriggan) [inline packages/bitcore-node/src/decorators/Loggify.ts]: Could be Math.max()

File: packages/bitcore-node/src/models/transaction.ts [hunk 2]
```diff
@@ -178,6 +183,27 @@ export class TransactionModel extends BaseModel<ITransaction> {
     }
   }
 
+  toMempoolSafeUpsert(
```
Comment (nitsujlangston) [inline packages/bitcore-node/src/models/transaction.ts]: the problem with mutating the ops in this separate method is that we now need to maintain both version of the op separately. Having a single op creator that dynamically structures the ops would prevent the two places from ever diverging their construction.

Comment (micahriggan) [reply packages/bitcore-node/src/models/transaction.ts]: Yeah, not ideal, but this only happens on a single mempool tx, so it should be light.

**Comment by matiu** [comment]
codeACK.

File: packages/bitcore-node/src/models/transaction.ts (hunk 2 above)
Comment (matiu) [reply packages/bitcore-node/src/models/transaction.ts]: Given the urgency of this fix, I would suggest we merge it like it is and refactor it later.

**Comment by matiu** [comment]
LGTM. I added an extra test here:
https://github.com/micahriggan/bitcore/pull/20

**Comment by micahriggan** [comment]
Merged the extra test

**Pull request merged**